src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,100 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elements;
+    private int size;
+
     @Override
     public void add(T value) {
-
+        if (elements == null) {
+            elements = (T[]) new Object[DEFAULT_CAPACITY];
+        }
+        if (size == elements.length) {
+            elements = Arrays.copyOf(elements, elements.length + elements.length / 2);
```

Comment:
Please create new class constant to give possibility to change growth coefficient